src/main/java/edu/wpi/grip/ui/preview/PreviewsView.java

```diff
@@ -48,24 +55,106 @@ public PreviewsView(EventBus eventBus) {
     }
 
     @Subscribe
+    /**
+     * This function is called when a preview button is pushed/triggered
+     */
     public synchronized void onSocketPreviewChanged(SocketPreviewChangedEvent event) {
-        Platform.runLater(() -> {
-            final OutputSocket<?> socket = event.getSocket();
-
-            if (socket.isPreviewed()) {
-                // If the socket was just set as previewed, add it to the list of previewed sockets and add a new view for it.
-                if (!this.previewedSockets.contains(socket)) {
-                    this.previewedSockets.add(socket);
-                    this.previewBox.getChildren().add(SocketPreviewViewFactory.createPreviewView(this.eventBus, socket));
+        Platform.runLater(() -> {//Run this function on the main gui thread
+
+            final OutputSocket<?> socket = event.getSocket(); //The socket whose preview has changed
```

Comment:
1. When commenting code, try to put comments on a separate line.  Otherwise, lines are much longer, and editing code is harder.
2. It would probably be better to have a few comments explaining what a block of code does than a comment translating each individual line.  We can all read java, so we don't need a line-for-line explanation, but a high-level description would be useful.